nibabel/ecat.py

```diff
@@ -155,10 +155,10 @@
 _dtdefs = ( # code, name, equivalent dtype
     (1, 'ECAT7_BYTE', np.uint8),
     (2, 'ECAT7_VAXI2', np.int16),
-    (3, 'ECAT7_VAXI4', np.float32),
+    (3, 'ECAT7_VAXI4', np.int32),
     (4, 'ECAT7_VAXR4', np.float32),
     (5, 'ECAT7_IEEER4', np.float32),
-    (6, 'ECAT7_SUNI2', np.uint16),
+    (6, 'ECAT7_SUNI2', np.int16),
```

Comment:
For example, ECAT7_SUNI2 does get read in as short int:

http://www.turkupetcentre.net/software/libdoc/libtpcimgio/ecat7r_8c_source.html#l00663